### PR TITLE
feat: add automated link checking

### PR DESCRIPTION
**Pull request by coolguy1771**
Signed-off-by: Tyler Witlin <45468117+coolguy1771@users.noreply.github.com>

## Description

Automated checking for broken links inside of documentation. Requires a token to be set called ``GITHUB_TOKEN`` to be set inside of the repository secrets. Alternatively, a bot can be created to work with GH actions across all repositories.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Technical improvement or removal of technical debt. 
- [ ] Minor change: comments, documentation, trivial fixes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [ ] Unit Tests

Optional:

- [ ] Functional Tests
- [ ] Integration Tests

### CODE REVIEW

File: .github/workflows/link-checker.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          state: open
+          labels: |
+            link-checker
+      - name: Update Issue
+        uses: peter-evans/create-issue-from-file@v3
+        with:
+          title: Broken links detected in docs 🔗
+          issue-number: "${{ steps.link-checker-issue.outputs.issue_number }}"
+          content-filepath: ./lychee.md
```
Comment (conneryn) [inline .github/workflows/link-checker.yaml:31]: Looks like this isn't actually the right path?
image


This is a guess:

```suggestion
          content-filepath: ./lychee/out.md
```

File: .github/workflows/link-checker.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.3.2
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        with:
+          args:  --exclude-all-private './content/**/*.md' -u 'Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0' --exclude http:* mongodb:*
```
Comment (conneryn) [inline .github/workflows/link-checker.yaml:21]: Is it possible to get the excludes defined in a config file (similar to the `.lycheeignore`)?  If not, no worries.

Comment (conneryn) [inline .github/workflows/link-checker.yaml:21]: Excluding all of `http:` feels a little too broad.  Is it possible to be a little more specific?  `--exclude-all-private` should already cover the localhost cases, and we could explicitly ignore any sample-project links... what other cases need to be excluded?

```suggestion
          args:  --exclude-all-private './content/**/*.md' -u 'Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0' --exclude http://sample-project-* mongodb:*
```

**Review by conneryn — APPROVED** [review]
Two thoughts, but I think good to go if we want to start trying these out.

File: .github/workflows/link-checker.yaml (hunk 2 above)
Comment (coolguy1771) [reply .github/workflows/link-checker.yaml]: It might be possible but I'm not sure that you can do it in a GH action.

**Pull request merged**